src/main/java/stsjorbsmod/actions/TimeEddyAction.java

```diff
@@ -66,7 +66,8 @@ public void update() {
         forEachApplicableMonsterPower(p -> p.atStartOfTurnPostDraw());
 
         // Monster turn
-        forEachApplicableMonsterPower(p -> p.duringTurn());
+        // This causes any actions added to the bottom as part of duringTurn to occur after all the others. Maybe not a big deal
+        addToBot(new ConsumerGameAction<>(o -> forEachApplicableMonsterPower(p -> p.duringTurn()), null));
```

Comment:
I'm not super comfortable with this approach; even if we were to update the rest of the lines to be more consistent, that would risk new problems with the second TimeEddyAction (in that any actions added to bottom by one of the callbacks would start running after the second TImeEddyAction runs again).

Reply:
Yeah, I agree. I'm not terribly happy about this approach either. It feels incredibly jank.

Reply:
Another idea I had was to patch `AbstractMonster::die` to remove actions from the action manager where `this` as the owner